src/api/plugins/tdbp/warren_tdbp/indicators.py

```diff
@@ -0,0 +1,550 @@
+    ):
+        """Initialize sliding window indicator."""
+        if until is None:
+            until = date.today()
+
+        super().__init__(
+            course_id=course_id,
+            until=until,
+            sliding_window_min=sliding_window_min,
+            active_actions_min=active_actions_min,
+            dynamic_cohort_min=dynamic_cohort_min,
+        )
+
+    @property
+    def xi_client(self):
+        """Retun AsyncClient instance for Experience Index queries."""
+        return AsyncClient(base_url=settings.BASE_XI_URL)
+
+    @property
+    async def course_actions(self) -> List[str]:
+        """Return actions related to course read from Experience Index."""
+        async with self.xi_client as client:
+            response = await client.get(f"/experiences/{self.course_id}")
+
+        relations = [
+            json.loads(relation) for relation in response.json()["relations_target"]
+        ]
+
+        return [relation["target_id"] for relation in relations]
+
+    def get_lrs_query(self):
+        """Construct the LRS query for statements which object is the course.
+
+        WARNING: this method is used only for key cache computing as the LRS
+        query requires to look for statements which object is related to the course.
+        """
+        return LRSQuery.construct(
+            query={
+                "activity": self.course_id,
+                "until": datetime.combine(self.until, time.min).isoformat(),
+            }
+        )
+
+    def _get_lrs_query(self, activity):
+        """Return LRS query for a course-related activity."""
+        return LRSQuery.construct(
+            query={
+                "activity": activity,
+                "until": datetime.combine(self.until, time.min).isoformat(),
+            }
+        )
+
+    @property
+    async def statements(self) -> pd.DataFrame:
+        """Return LRS statements related to course actions."""
+        raw_statements = pd.DataFrame()
+        for action_id in await self.course_actions:
+            try:
+                data = [
+                    value
+                    async for value in self.lrs_client.read(
+                        target=self.lrs_client.statements_endpoint,
+                        query=self._get_lrs_query(activity=action_id),
+                    )
+                ]
+            except BackendException as exception:
+                raise LrsClientException("Failed to fetch statements") from exception
+
+            raw_statements = pd.concat([raw_statements, pd.json_normalize(data)])
+
+        raw_statements["timestamp"] = pd.to_datetime(raw_statements["timestamp"])
+        raw_statements["date"] = raw_statements["timestamp"].dt.date
+
+        # Check whether statements are distributed at least over the sliding window
+        statements_window = self.until - raw_statements["date"].min()
+        if statements_window.days < self.sliding_window_min:
+            raise Exception(
```

Comment:
Maybe we should define a more specific, project-related error?

```suggestion
            raise ConfigurableIndicatorConsistencyError(
```